parse/DoubleValueRefParser.cpp

```diff
@@ -117,25 +120,39 @@ parse::double_parser_rules::double_parser_rules(
         =   int_complex_grammar [ _val = construct_movable_(new_<ValueRef::StaticCast<int, double>>(deconstruct_movable_(_1, _pass))) ]
         ;
 
+    named_real_valueref
+        = (     tok.Named_ >> tok.Real_
```

Comment:
are both `NamedReal` (and variants) and `Lookup` necessary to support? Ideally there'd be one way to referenced a named ValueRef that works everywhere... or a consistent set of type-dependent ways... though needing the scripter to decide on a type isn't ideal.

Reply:
The different variants are there because the parser ran into the wrong parse paths. Is there a FOCS valueref which you could use to force/cast a type?

Before that the name-in-the-middle were called `Named` and that did not work so well. Then the name-in-the-middle ones got typed an and `Lookup` was also called `Named` and that also did not work so well.

I restructured a lot and i have more working knowledge of the parser library so it might work now, will try.

Reply:
> Is there a FOCS valueref which you could use to force/cast a type?

not sure what you mean, but https://github.com/freeorion/freeorion/blob/master/universe/ValueRefs.h#L201 exists...

Reply:
i meant if it is available in FOCS

Reply:
same answer...